src/engraving/automation/internal/automation.cpp

```diff
@@ -58,43 +60,91 @@ const AutomationCurve& Automation::curve(const muse::ID& staffId) const
     return it->second;
 }
 
-void Automation::addPoint(const muse::ID& staffId, int utick, const AutomationPoint& p)
+void Automation::addPoint(const muse::ID& staffId, int utick, AutomationPoint& point)
 {
-    auto it = m_curveMap.find(staffId);
-    IF_ASSERT_FAILED(it != m_curveMap.end()) {
+    auto curveIt = m_curveMap.find(staffId);
+    IF_ASSERT_FAILED(curveIt != m_curveMap.end()) {
+        return;
+    }
+
+    AutomationCurve& curve = curveIt->second;
+    AutomationPoints& points = curve.points[utick];
+
+    point.id = ID();
+
+    for (auto it = curve.points.rbegin(); it != curve.points.rend(); ++it) {
+        for (const AutomationPoint& p : it->second) {
+            point.id = std::max(point.id, p.id + 1);
+        }
+    }
+
+    if (!point.id.isValid()) {
+        point.id = 1;
+    }
+
+    if (points.empty()) {
+        points.push_back(point);
         return;
     }
 
-    AutomationCurve& curve = it->second;
-    curve.points[utick] = p;
+    // Keep points sorted by value
```

Comment:
Doesn't that mean we won't be able to distinguish between the following two curves?
<img width="200" src="https://github.com/user-attachments/assets/ae2afe2e-e624-4697-8fbe-a9bc0cefc2cd"/>

Maybe instead of allowing an arbitrary number of points per tick, it makes more sense to allow only one point, but let it have a "before" and "after" value.